lldb/include/lldb/Core/Architecture.h

```diff
@@ -109,6 +110,24 @@ class Architecture : public PluginInterface {
   virtual const MemoryTagManager *GetMemoryTagManager() const {
     return nullptr;
   }
+
+  // This returns true if a write to the named register should cause lldb to
+  // reconfigure its register information. For example on AArch64 writing to vg
+  // to change the vector length means lldb has to change the size of registers.
+  virtual bool RegisterWriteCausesReconfigure(const char *name) const {
```

Comment:
Why a `const char*` and not a `StringRef`? That would get rid of the `strcmp` in the AArch64 implementation.

Reply:
+1

Reply:
Done.